src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +88,163 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
         });
     };
 
-    @action handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
+    handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
         this.validator = jsonSchema ? ajv.compile(jsonSchema) : undefined;
         this.pathsByTag = {};
 
-        this.schema = schema;
-        this.addMissingSchemaProperties();
-        this.setSchemaLoading(false);
+        this.removeObsoleteData(this.schema, schema, this.data);
+        this.loadAndMergeOriginData(this.schema, schema).then(action(()=> {
+            this.schema = schema;
+            this.addMissingSchemaProperties();
+            this.setSchemaLoading(false);
+        }));
     };
 
+    loadAndMergeOriginData(oldSchema: Schema, newSchema: Schema) {
+        // origin data must be loaded only when switching back to the origin template
+        if (oldSchema && toJS(this.type) === this.data['originTemplate']) {
+            return this.resourceStore.requestData().then((data: Object) => {
+                const result = this.calculateDifference(oldSchema, newSchema, this.data, data);
```

Comment:
do we need to calculate the difference here? could we return the merged data instead of just the difference? 🤔 
that way we would not need the `setMultipleRecursive` method

Reply:
yes that is also possible, seems like I misunderstood you the last time, I thought you specifically wanted to calculate only the difference 😅